prompt2model/model_trainer/generate.py

```diff
@@ -236,13 +247,164 @@ def tokenize_dataset(
         }
         return datasets.Dataset.from_dict(preprocessed_dict)
 
+    def search_best_hyperparameters(
+        self,
+        hyperparameters: dict[str, Any],
+        training_datasets: Dataset,
+        validation_datasets: Dataset,
+    ) -> transformers.Trainer:
+        """Method to do automatic hyperparameter search.
+
+        The hyperparameters in this argument will be the same as the hyperparameter
+        choices in the `train_model()` method. However the hyperparameter_choices
```

Comment:
I don't think this is entirely accurate, since we're also adding the `optuna` parameters, which are otherwise not given to the `train_model` method. Can you clarify this in the comment?

Reply:
It is coming from the change in the schema that I did. Since we have `static_parameters` and `optuna` as keys, so my thought process that to pass that schema while searching hyperparameters. 

Another reason was that I wanted to search for hyperparameters in the `train_model` method only where the process of optimization will be dependent on how the user provides the schema. If `optuna` is present then we will go for the search else not.